specs/HubProtocol.md

```diff
@@ -348,8 +363,160 @@ World;0:;
 
 Note that the final frame still ends with the `;` terminator, and that since the body may contain `;`, newlines, etc., the length is specified in order to know exactly where the body ends.
 
+## MessagePack (MsgPack) encoding
+
+In the MsgPack Encoding of the SignalR Protocol, each Message is represented as a single MsgPack array containing items that correspond to properties of the given hub protocol message. The array items may be primitive values, arrays (e.g. method arguments) or objects (e.g. argument value). The first item in the array is the message type.
```

Comment:
> The benefit of the self describing format is that certain types of changes don't require a version bump (like additing properties).
Agreed but one of the main objectives of the binary format was efficiency/small messages

> It would be trivial since we have a layer that turns the protocol format into messages.
Yes it wouldn't be hard.

> I just wanted to understand the general approach of using arrays vs objects since msgpack supports both.
I guess it felt natural to use objects when using JSON and it was easy to implement. I was kind of fixed on this approach when starting with MsgPack but then I realized using arrays will be more efficient so switched to this.